sql/catalyst/src/main/scala/org/apache/spark/sql/types/numerics.scala

```diff
@@ -121,8 +121,8 @@ object FloatExactNumeric extends FloatIsFractional {
   private def overflowException(x: Float, dataType: String) =
     throw new ArithmeticException(s"Casting $x to $dataType causes overflow")
 
-  private val intUpperBound = Int.MaxValue.toFloat
-  private val intLowerBound = Int.MinValue.toFloat
+  private val intUpperBound = Int.MaxValue
+  private val intLowerBound = Int.MinValue
   private val longUpperBound = Long.MaxValue.toFloat
   private val longLowerBound = Long.MinValue.toFloat
```

Comment:
seems we can remove `toFloat` here too? also the `toDouble` in `DoubleExactNumeric`. They will be promoted anyway.

Reply:
Agree. It looks more gracefully with consistent style.